sample-graphs/update.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -eu
+
+SOURCECRED_REMOTE="git@github.com:sourcecred/sourcecred.git"
+
+# Taken from https://stackoverflow.com/a/246128
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Use a clean SOURCECRED_DIRECTROY to avoid cache pollution.
+# Maybe reconsider this if we start storing really large projects here
+# (but putting really large projects' graphs in this Git repository is likely
+# a bad idea anyway).
+export SOURCECRED_DIRECTORY="$(mktemp -d --suffix ".sourcecred-directory")"
+
+# Just clone a fresh copy of SourceCred for convenience.
+sourcecred_repo="$(mktemp -d --suffix ".sourcecred-repo")"
+git clone "${SOURCECRED_REMOTE}" "${sourcecred_repo}"
+cd "${sourcecred_repo}"
+
+yarn && yarn backend
+
+for_repo () {
+  node bin/sourcecred.js load "$1"/"$2"
+  node bin/sourcecred.js export-graph "$1"/"$2" > "$DIR/$1_$2.json"
+}
+
+for_repo "sourcecred" "sourcecred"
+for_repo "sourcecred" "research"
```

Comment:
It’s a bit strange to commit the sourcecred/research graph into
sourcecred/research. It becomes impossible for the graph to ever be
correct as soon as it’s checked in, because really there’s a cycle in
the dependency graph… thoughts?

Reply:
For someone working on research, the `sourcecred/research` graph may well be the one they have the best intuition on, so including it is valuable. I think that defining the graph as "correct" if it perfectly matches the current state of the repository is not reasonable, since I don't expect this update script to be run more often than (say) once every week or two, so the graphs will almost never be perfectly synchronized.

Reply:
Yep, those are reasonable thoughts, and approximately what I expected to
hear. :-)

The only remaining pitfall, then, is if the graph itself becomes
self-referential—e.g., SourceCred adds file contents to the graph in
some sense such that the act of regenerating the graph and committing it
materially bloats the size of the repository. With our current approach
to graphs, this doesn’t seem likely, so it’s nothing more than something
to keep in mind.